pkg/stanza/operator/parser/json_array/json_array_parser.go

```diff
@@ -0,0 +1,127 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+package json_array // import "github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/operator/parser/json_array"
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"go.opentelemetry.io/collector/featuregate"
+	"go.uber.org/zap"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/entry"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/operator"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/operator/helper"
+
+	"github.com/valyala/fastjson"
+)
+
+const operatorType = "json_array_parser"
+
+var jsonArrayParserFeatureGate = featuregate.GlobalRegistry().MustRegister(
+	"filelog.jsonParserArray",
```

Comment:
Technically this is available for more receivers as well. Also not sure the best name but I think this is sufficient.
```suggestion
	"logs.jsonParserArray",
```